webui/react/src/pages/ExperimentList.tsx

```diff
@@ -80,13 +81,126 @@ const ExperimentList: React.FC = () => {
     },
   );
   const initSorter = storage.getWithDefault(STORAGE_SORTER_KEY, { ...defaultSorter });
-  const [ pagination, setPagination ] = useState<Pagination>({ limit: initLimit, offset: 0 });
-  const [ total, setTotal ] = useState(0);
+  const [ experiments, setExperiments ] = useState<ExperimentItem[]>();
   const [ filters, setFilters ] = useState<ExperimentFilters>(initFilters);
-  const [ sorter, setSorter ] = useState(initSorter);
+  const [ isUrlParsed, setIsUrlParsed ] = useState<boolean>(false);
+  const [ pagination, setPagination ] = useState<Pagination>({ limit: initLimit, offset: 0 });
   const [ search, setSearch ] = useState('');
-  const [ experiments, setExperiments ] = useState<ExperimentItem[]>();
   const [ selectedRowKeys, setSelectedRowKeys ] = useState<string[]>([]);
+  const [ sorter, setSorter ] = useState(initSorter);
+  const [ total, setTotal ] = useState(0);
+
+  /*
+   * When filters changes update the page URL.
+   */
+  useEffect(() => {
+    if (!isUrlParsed) return;
+
+    const searchParams: string[][] = [
+      [ 'archived', filters.showArchived ? '1' : '0' ],
+      [ 'limit', pagination.limit.toString() ],
+      [ 'offset', pagination.offset.toString() ],
+      [ 'sortDesc', sorter.descend ? '1' : '0' ],
+    ];
```

Comment:
**blocker:** I believe all of the params need to be set with a default of at least an empty string `''`. For example if `username` is only set when `filters.username` is available, then in the URL query string, `&username=` will not be available. And when that link is copied or shared, if Jane pasting the URL has a user preference of `username` set to `jane`, the URL which should show All users, will default to only showing `jane`.

Then on the parsing end when you do a `params.get('username')` you should get an empty string instead of a `null` to know that it's set to All as opposed to using the user preference default. Since we are taking the URL params as priority #1 followed by user preferences, then default settings and values.

Reply:
so instead of
```
    if (filters.username) {
      searchParams.push([ 'username', filters.username ]);
    }
```
It would be something more like:
```
    const searchParams: string[][] = [
      ...
      [ 'username', filters.username || '' ],
    ];
```

Reply:
I noticed that you had `URLSearchParams` as a record originally before you changed to array of an array. What changed your mind to this format? Just curious.

Reply:
1. I missed to handle the case for the `username`, was already there for `label`. Pushed a fix for this.
2. TS interface for `URLSearchParams` accept `string[][] | Record<string, string> | string | URLSearchParams`. Since we have the `label` params that needs to accept multiple values `Record` would have no longer worked. That's why I migrate to a `string[][]` array.